src/pages/home/report/ReportActionCompose.js

```diff
@@ -318,15 +319,19 @@ class ReportActionCompose extends React.Component {
      */
     addEmojiToTextBox(emoji) {
         const emojiWithSpace = `${emoji} `;
-        const newComment = this.comment.slice(0, this.state.selection.start)
-            + emojiWithSpace + this.comment.slice(this.state.selection.end, this.comment.length);
-        this.setState(prevState => ({
-            selection: {
-                start: prevState.selection.start + emojiWithSpace.length,
-                end: prevState.selection.start + emojiWithSpace.length,
-            },
-        }));
-        this.updateComment(newComment);
+        const newComment = this.comment.slice(0, this.selection.start)
+            + emojiWithSpace + this.comment.slice(this.selection.end, this.comment.length);
+        this.selection = {
+            start: this.selection.start + emojiWithSpace.length,
+            end: this.selection.start + emojiWithSpace.length,
+        };
+
+        this.textInput.updateSelection(this.selection);
+
+        // this will call the function we passed
+        // to the TextInput's onChangeText,
+        // so updateComment gets called after this.
```

Comment:
```suggestion
        // This will call the function we passed to the TextInput's onChangeText.
        // So updateComment gets called after this.
```